File: InternetAuction.API/Client/src/app/core/models/user.model.ts
import {Lot} from "./lot.model";
import {Bid} from "./bid.model";
import {Image} from "./image.model";

export interface User {
    id: number,
    userName: string,
    firstName: string,
    lastName: string,
    balance: string,
    profileImage: Image,
    registeredLots: Lot[],
    boughtLots: Lot[],
    bids: Bid[]
}

File: InternetAuction.BLL/Models/User/UserModel.cs
using System.Collections.Generic;
using InternetAuction.BLL.Models.Bid;
using InternetAuction.BLL.Models.Image;
using InternetAuction.BLL.Models.Lot;

namespace InternetAuction.BLL.Models.User
{
    /// <summary>
    /// Represents a user DTO.
    /// </summary>
    public class UserModel
    {
        public int Id { get; set; }

        public string UserName { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public decimal Balance { get; set; }

        public ImageModel ProfileImage { get; set; }

        public List<LotModel> RegisteredLots { get; set; }

        public List<LotModel> BoughtLots { get; set; }

        public List<BidModel> Bids { get; set; }
    }
}